GPU/GPUTracking/Global/GPUChainTrackingCompression.cxx

```diff
@@ -208,11 +206,125 @@ int GPUChainTracking::RunTPCCompression()
 int GPUChainTracking::RunTPCDecompression()
 {
 #ifdef GPUCA_HAVE_O2HEADERS
+  // mRec->PushNonPersistentMemory(qStr2Tag("TPCDCMPR"));
+  ClusterNativeAccess* original = new ClusterNativeAccess;
+  original->clustersLinear = new ClusterNative[mIOPtrs.clustersNative->nClustersTotal];
+  memcpy((void*)original->clustersLinear, mIOPtrs.clustersNative->clustersLinear, mIOPtrs.clustersNative->nClustersTotal * sizeof(mIOPtrs.clustersNative->clustersLinear[0]));
+  memcpy((void*)original->nClusters, mIOPtrs.clustersNative->nClusters, NSLICES * GPUCA_ROW_COUNT * sizeof(mIOPtrs.clustersNative->nClusters[0][0]));
+  original->setOffsetPtrs();
+
+  RecoStep myStep = RecoStep::TPCDecompression;
+  bool doGPU = GetRecoStepsGPU() & RecoStep::TPCDecompression; // with -g gives true
+  GPUTPCDecompression& Decompressor = processors()->tpcDecompressor;
+  GPUTPCDecompression& DecompressorShadow = doGPU ? processorsShadow()->tpcDecompressor : Decompressor;
   const auto& threadContext = GetThreadContext();
+
+  CompressedClusters cmprClsHost = *mIOPtrs.tpcCompressedClusters;
+  CompressedClusters& inputGPU = Decompressor.mInputGPU;
+  inputGPU.nAttachedClusters = cmprClsHost.nAttachedClusters;
+  inputGPU.nUnattachedClusters = cmprClsHost.nUnattachedClusters;
+  inputGPU.nTracks = cmprClsHost.nTracks;
+  inputGPU.nAttachedClustersReduced = inputGPU.nAttachedClusters - inputGPU.nTracks;
+  inputGPU.nSliceRows = NSLICES * GPUCA_ROW_COUNT;
+  inputGPU.nComppressionModes = param().rec.tpc.compressionTypeMask;
+  inputGPU.solenoidBz = param().bzkG;
+  inputGPU.maxTimeBin = param().par.continuousMaxTimeBin;
+  SetupGPUProcessor(&Decompressor, true);
+
+  size_t copySize = AllocateRegisteredMemory(Decompressor.mMemoryResInputGPU);
+  WriteToConstantMemory(myStep, (char*)&processors()->tpcDecompressor - (char*)processors(), &DecompressorShadow, sizeof(DecompressorShadow), 0);
+
+  int outputStream = 0;
+  bool toGPU = true;
+  CompressedClusters& inputGPUShadow = DecompressorShadow.mInputGPU;
+  SynchronizeStream(outputStream);
```

Comment:
You do not need to synchrinize with this stream I believe? Nothing is happening in this stream before.